templates/commands/render.go

```diff
@@ -353,6 +349,80 @@ func (c *RenderCommand) checkUnknownInputs(spec *model.Spec) []string {
 	return unknownInputs
 }
 
+// resolveInputs combines flags, user prompts, and defaults to get the full set
+// of template inputs.
+func (c *RenderCommand) resolveInputs(ctx context.Context, spec *model.Spec) error {
+	if unknownInputs := c.checkUnknownInputs(spec); len(unknownInputs) > 0 {
+		return fmt.Errorf("unknown input(s): %s", strings.Join(unknownInputs, ", "))
+	}
+
+	if c.flags.Prompt {
+		isATTY := (c.Stdin() == os.Stdin && isatty.IsTerminal(os.Stdin.Fd()))
+		if !isATTY {
+			return fmt.Errorf("the flag --prompt was provided, but standard input is not a terminal")
+		}
+
+		if err := c.promptForInputs(ctx, spec); err != nil {
+			return err
+		}
+	} else {
+		c.collapseDefaultInputs(spec)
+		if missing := c.checkInputsMissing(spec); len(missing) > 0 {
+			return fmt.Errorf("missing input(s): %s", strings.Join(missing, ", "))
+		}
+	}
+
+	return nil
+}
+
+// promptForInputs looks for template inputs that were not provided on the
+// command line and prompts the user for them. This mutates c.flags.Inputs.
+//
+// This must only be called when the user specified --prompt and the input is a
+// terminal (or in a test).
```

Comment:
nit: consider copy the example in the PR description over :)

Reply:
Will do in a followup so I don't lose my approval bit.